lib/base_service.ts

```diff
@@ -52,18 +53,19 @@ export interface Credentials {
   password: string;
   api_key: string;
   url: string;
+  access_token: string;
 }
 
 function hasCredentials(obj: any): boolean {
-  return obj && ((obj.username && obj.password) || obj.api_key);
+  return obj && ((obj.username && obj.password) || obj.api_key || obj.access_token);
 }
 
 function hasBasicCredentials(obj: any): boolean {
   return obj && obj.username && obj.password;
 }
 
-function acceptsApiKey(name: string): boolean {
```

Comment:
why does this need to be removed? im wondering if this reduces functionality in some cases for when we check visrec

Reply:
That function was only being used to determine if an error statement should say "pass in an api_key". I removed that error statement because when supporting more authentication methods (api_key, iam_api_key, access_token, refresh_token, etc) it will get too complicated to be worth printing specific error statements. So I removed that function as well